### ISSUE TITLE
[REQ] Multiple query tabs within same connection

### ISSUE THREAD

**Issue by ghost** [sequelpro/sequelpro#1069]
Issue imported from Google Code: https://code.google.com/p/sequel-pro/issues/detail?id=1069
Reporter:Sajaki9@gmail.com Date:2011-06-03 23:54:15 Status:Accepted

there is only one query screen. to make another query i need to open another session window. i'd like to have multiple query windows in a session, like in TOAD. 

What version of Sequel Pro are you using? 

version 0.9.8.1

**Comment by Den4ik** [comment]
+1

**Comment by coladou** [comment]
+1

**Comment by DaHaiz** [comment]
+1

**Comment by TomKeur** [comment]
+1

**Comment by clkr** [comment]
+1

**Comment by evivz** [comment]
+1

**Comment by standingwave** [comment]
+1

**Comment by dhmmasson** [comment]
+1

**Comment by dbeja** [comment]
+1

**Comment by calebeaires** [comment]
+1

**Comment by ederlindsay** [comment]
+1

**Comment by rdiazv** [comment]
+1

**Comment by rafaelugaz** [comment]
+1

**Comment by ggmichaelgo** [comment]
+1

**Comment by hector-del-rio** [comment]
+1

**Comment by cykirsch** [comment]
+1

**Comment by zastil** [comment]
+1

**Comment by jgliner** [comment]
+1

**Comment by kevin-stafford** [comment]
+1

**Comment by sebdec** [comment]
+1

**Comment by renatogcarvalho** [comment]
+1

**Comment by PushEngineering** [comment]
+1 

Recently migrated from the now very broken (on a Mac at least) MySQL Workbench. Query tabs is the only feature we're really missing...

**Comment by hokey** [comment]
👍

**Comment by cristian0** [comment]
+1

**Comment by graemian** [comment]
+1

**Comment by tsm91** [comment]
http://nightly.sequelpro.com/ is down, where can i get the nighty version?

**Comment by abhibeckert** [comment]
@tsm91 we changed the URL to remove the subdomain as part of switching to SSL.

It's now: https://sequelpro.com/nightly/

**Comment by adrianthedev** [comment]
+1

**Comment by axelf** [comment]
+1

More than 3 years ago the topic was opened and now... we are still missing this feature

**Comment by adrianthedev** [comment]
You might check out [DBNinja](http://www.dbninja.com/). It's web based and development on the project halted a while back, but I use it for a few years now specifically for the tabbed interface. A real productivity booster.

**Comment by dmoagx** [comment]
From what I've seen, people referring to "Query tabs" are actually talking about at least 3 very different concepts.

If people would stop annoying everyone with the useless "+1" spam and instead actually describe what the current tab implementation is lacking and what they imagine with "Query tabs" this may see more interest from developers...

**Comment by adrianthedev** [comment]
You might be right @dmoagx. I know I find a lot of value in "tabs as views" implementation like in DB Ninja.
One use case would be: an action is performed in the app and data updated in two or more tables. I might have a tab in the `users` table to follow info over there, one tab in the `user_image` table, one in `images` table and a different tab with a custom query. After the action has performed I would just browse through them and just refresh each view.
Huge productivity boost!

**Comment by axelf** [comment]
@dmoagx what do you mean with 3 different concepts? This concept exists in other MySQL-clients for years and they work exactly the same (heidiSql, ninjadb or toad). The lack is that the tab implemantation does not exist. The Problem was described in previous posts.

**Comment by PushEngineering** [comment]
Yep just echoing above posts - a single connection but multiple tabs to have multiple queries open and viewable at the same time, as @adrianthedev pointed out this is improves usability/productivity quite a lot! 

For people waiting for the feature; worth bearing in mind that this functionality is "shiv-able" by just opening multiple connection tabs to the same db server. It works OK but it's a bit clunky and inefficient in terms of holding open separate connections.

I always thought +1's were useful in assessing the actual weight of the demand for a feature - it's clearly the main thing people would love to see :)

**Comment by dmoagx** [comment]
So what @adrianthedev describes and what I can see from the DBNinja screenshots is 'tabs at the top level' where one tab would show table A in structure view and another tab would show table B in content view.
This is essentially the same as our connection tabs, which - again - I don't understand what our implementation lacks...

@axelf Sequel Pro has had tabs for years. Be more specific.

@PushEngineering Why is opening multiple connections clunky? An idle connection doesn't really do anything and with the way MySQL works you can never execute multiple queries in parallel on a single connection.

Also some people want to have "Multiple query tabs with a single results table" while others want "Multiple query tabs, each with its own results table" from what I understand.

**Comment by PushEngineering** [comment]
@dmoagx 

Inefficient in terms of open connections - some db servers limit these :)

Only clunky in that it's - open another db connection, select db server, select db, start querying. Rather than a single +new query tab click :)

FWIW, both of these are marginal improvements - but would make this great app just a little better.

**Comment by axelf** [comment]
I mean having multiple tabs for queries in the current connection tab. That is user friendly and much faster than starting a new connection tab.

**Comment by dmoagx** [comment]
@PushEngineering You can press alt+T to duplicate connection, which will look exactly like the existing one.

**Comment by abhibeckert** [comment]
The default connection limit is 151. Having two or three of them is not going to get you any closer than this.

I lean towards having a separate connection so you can run two queries at once, although this means session variables wouldn't be shared.

There are many ways to implement this feature and we will not implement all of them. So the most useful comments are those which explain a use case scenario where the current implementation is inefficient.

This will help us come up with a solution to solve problems for everyone. Right now, we are leaning towards having a "Query" items in the left sidebar, similar to tables/stored procedures/functions.

**Comment by PushEngineering** [comment]
Cmd +T still requires you to select db server + db again.

Connections sometimes are limited per user - on some servers I connect to I can only have one active connection, which effectively means no tabs :(

I think the request is legitimate - but I will leave it here now :)

**Comment by abhibeckert** [comment]
> Cmd +T still requires you to select db server + db again.

Option + T doesn't. In the latest builds (as of a few hours ago) we've made this more obvious and changed the shortcut to Cmd + Shift + T.

> Connections sometimes are limited per user - on some servers I connect to I can only have one active connection, which effectively means no tabs :(

Thanks, we'll keep that in mind.

**Comment by adrianthedev** [comment]
Yes! As @PushEngineering was mentioning [earlier](https://github.com/sequelpro/sequelpro/issues/1069#issuecomment-283777141) there are a few UX considerations.

**Comment by adrianthedev** [comment]
I would recommend using DB Ninja a bit to catch "the feeling" of the tabs. I think that would clarify a few things.

**Comment by jsd3** [comment]
Mysql Workbench, when it works, has tabs and they're essential to keeping scripts separate and work organized when things get busy.  I echo the need for this functionality in Querious, which otherwise seems like a decent tool so far.  Why not just add tabs for query management on a single connection since so many people want it?

**Comment by skyvondave** [comment]
It's ironic that this is being marked as stale, when I've stopped using Sequel Pro mostly because I can't have multiple query tabs per connection... it might otherwise be better than MySQL Workbench, I don't know, I got too frustrated at only being able to have one query tab per connection, and the hassle of jumping around multiple connection tabs to work with different queries was just too much.

I only use it now to see a moving import bar.

However, if this issue were actually worked on, it wouldn't be stale, would it?

**Comment by rohit-ambre** [comment]
I used to work on Linux before and recently started working on Mac, even though working on ubuntu I installed `heidi Sql` using wine because of the ease it offers while working on queries.
Now while working on Mac, my colleague asked me to try sequel pro and I did find it quite good, I find the search filter on Content tab is best thing but I just miss having multiple query tabs....
Now when am writing multiple and long queries for the same connections and same database why I need to open new window or open whole new tab after making new connection....I think there should be just `+` mark besides query tab which would one more query text editor because now I have to keep scrolling and selecting the exact query I want to run.
I think @PushEngineering  has already suggested the same things
and it also crashes the whole app after I close any tab and that's frustrating, I know that's a different issue

I will check `dbeaver` now and I'll also keep checking on sequel pro development

**Comment by jpenna** [comment]
My company doesn't allow more than 1 connection to the SQL server per user.

The work around I've found is using "Query Favorites" to switch between queries and for local development use Workbench.

**Comment by skyvondave** [comment]
I've switched to DBeaver... it seems much easier to work with, and allows multiple tabs.

Sequel Pro crashed a lot late last year on my Mac, even after fully upgrading everything, so I dumped it.

I've heard that Sequel Pro stopped getting much support recently, and DBeaver seems like it has much better stability, functionality, and support.
